File: solutions/julia/sieve/1/sieve.jl
function sieve(limit)
    s = [true for _ in 1:limit]
    s[1] = false
    for i in 2:limit
        if s[i]
            j = 2 * i
            while j <= limit
                s[j] = false
                j += i
            end
        end
    end
    [i for (i, j) in zip(1:limit, s) if j]
end

File: solutions/kotlin/sieve/1/src/main/kotlin/Sieve.kt
object Sieve {

    fun primesUpTo(upperBound: Int): List<Int> {
        require(upperBound >= 0)
        val list = Array<Boolean>(upperBound + 1) { true }
        list[0] = false
        if (upperBound >= 1) {
            list[1] = false
        }
        for (p in 2..upperBound) {
            if (list[p]) {
                for (q in ((2*p..upperBound).step(p))) {
                    list[q] = false
                }
            }
        }
        return list.zip(0..upperBound).filter {
            it.first
        }.map {
            it.second
        }
    }
}
